### PR TITLE
NEMI-461 Bulk load analyte

### PR DESCRIPTION
**Pull request by ajmccarthy**
- Add template for bulk upload page
- Add button in common/admin.py to direct to bulk upload page
- Add a view model and functions to process a csv file and upload the data 
- Validations and template will be handled in a later task

### CODE REVIEW

File: nemi/common/admin.py [hunk 1]
```diff
@@ -793,6 +828,78 @@ def has_delete_permission(self, request, *args, **kwargs):
...
+            file_data = csv_file.read().decode('utf-8')
+            csv_data = io.StringIO(file_data)
+            # Skip header
+            next(csv_data)
+            analyte_method_record = []
+            error_messages = []
+            row_index = 1
+            current_method_id = request.GET.get('method_id','')
+            for row in csv.reader(csv_data):
+                print (f'current method id after request.get{current_method_id}')
```
Comment (mbucknell) [inline nemi/common/admin.py:889]: Please remove print statements

File: nemi/nemi_project/templates/admin/bulk_upload.html [hunk 2]
```diff
@@ -0,0 +1,65 @@
+{% extends "admin/base.html" %}
+{% load static %}
+
+{% block title %}NGWMN Well Registry Administration{% endblock %}
```
Comment (mbucknell) [inline nemi/nemi_project/templates/admin/bulk_upload.html:4]: Whoops! Needs to be changed to the appropriate title value for NEMI.

File: nemi/nemi_project/templates/admin/bulk_upload.html [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+                        {% for key, value in error.items %}
+                            <li>{{ key }}: {{ value }}</li>
+                        {% endfor %}
+                    </ol>
+                </li>
+            {% endfor %}
+        </ul>
+    {% endif %}
+
+    {% if warnings %}
```
Comment (mbucknell) [inline nemi/nemi_project/templates/admin/bulk_upload.html:33]: You aren't creating any warnings and this piece of the code in NGWMN is misused. Please remove.

File: nemi/nemi_project/templates/admin/bulk_upload.html [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    <form action="." method="post" enctype="multipart/form-data">
+    {% csrf_token %}
+        <div class="usa-form-group">
+            {% if file_error %}
+            <ul class="errorlist">
+
+                <li>{{ file_error }}</li>
+            </ul>
+            {% endif %}
+            <label class="usa-label font-body-sm" for="{{ form.file.id_for_label}}">Choose csv file containing method analytes</label>
```
Comment (mbucknell) [inline nemi/nemi_project/templates/admin/bulk_upload.html:58]: The usa-label  (and usa-file-input, below) class is only useful if we are using USWDS design system which we are not for NEMI. This should be removed. Similarly 'font-body-sm' is a USWDS class and is not needed here

**Review by mbucknell — APPROVED** [review]
Several suggestions on cleaning up the code.

File: nemi/common/admin.py [hunk 5]
```diff
@@ -7,16 +7,27 @@
...
 from django.forms.models import BaseInlineFormSet
+from django.forms import ModelForm, FileField, Form
+from django.views.generic import View
 from django.http import HttpResponseRedirect
 from django.template.defaultfilters import slugify
 from django.urls import reverse
+from django.urls import path
+from django.shortcuts import render
 import PyPDF2
+import io
```
Comment (mbucknell) [inline nemi/common/admin.py:18]: Please put the imports in the order recommended by Pep 8 (https://www.python.org/dev/peps/pep-0008/#imports).  The following three imports are standard library imports. Within a group, please alphabetize imports.

File: nemi/common/admin.py [hunk 6]
```diff
@@ -610,6 +628,18 @@ def submit_for_review(self, request, queryset):
     submit_for_review.label = 'Submit for review'
     submit_for_review.short_description = 'Submit method for review'
 
+    def analyte_bulk_load(self, request, queryset):
+        # Build URL to redirect to from 'Bulk upload Analytes' button. Pass in method_id.
```
Comment (mbucknell) [inline nemi/common/admin.py:632]: Comment is confusing "Build URL to redirect **to from** 'Bulk upload Analytes' button"

**Review by mbucknell — APPROVED** [review]
Some minor nits

File: nemi/common/admin.py (hunk 5 above)
Comment (ajmccarthy) [reply nemi/common/admin.py:18]: I didn't know this, thanks.

File: nemi/common/admin.py (hunk 6 above)
Comment (ajmccarthy) [reply nemi/common/admin.py:632]: Good point.

**Pull request merged**